File: python/secret-handshake/secret_handshake.py
ACTIONS = ['wink', 'double blink', 'close your eyes', 'jump', 'reverse']


def commands(code):
    code = int(code, 2)
    actions = []
    for i, a in enumerate(ACTIONS):
        if code & (1 << i) != 0:
            actions.append(a)
    if 'reverse' in actions:
        actions = actions[-2::-1]
    return actions

File: julia/secret-handshake/secret-handshake.jl
function secret_handshake(code::Integer)
    actions = [ "wink", "double blink", "close your eyes", "jump" ]

    f = code & 0b1_0000 ≠ 0 ? reverse : identity

    # first take, looping
    ##handshake = []
    ##for (i, action) in enumerate(actions)
    ##    key = 1 << (i - 1)
    ##    if code & key > 0
    ##        push!(handshake, action)
    ##    end
    ##end

    # second  take, iterables
    ##handshake = map(
    ##    i -> actions[i],
    ##    filter(
    ##        i -> code & (1 << (i-1)) > 0,
    ##        eachindex(actions)
    ##    )
    ##)

    # third take, comprehensions
    handshake =  [
        action
        for (i, action) in enumerate(actions)
        if code & (1 << (i-1)) ≠ 0
    ]

    f(handshake)
end
